### DISCUSSION TITLE
Give a way to monitor http request from an adapater

### DISCUSSION THREAD

**Discussion by mathieu-pousse** [keycloak/keycloak#8686] [Ideas]
Hello, 

I am wondering if you will agree that we enhance the `HttpClientBuilder` used in the Java adapters, to propose a simple and efficient way to monitor the http calls made by the library.

I'd like, as a developper using the keycloak adapter, to have a way to monitor the number / duration of calls made to get token refresh tokens, ...
We had an issue with our Keycloak server and could have handle that on the client side with caches / circuit breaker / fallback... if we could have had any monitoring.

This could be easily achieved by registering an `HttpRequestInterceptor` and `HttpResponseInterceptor`.

Those interceptors could be used by a developper to add logging, debugging, metrics, ...

WDYT ?

My idea was to add a few more fields to the `AdapterConfig` and simply propagate them in the `HttpClientBuilder`

Let me know

I'll we be pleased to contribute.

**Comment by stianst** [comment]
Rather than create something bespoke here, and I think it would be better to look at tracing support, such as Open Telemetry.
